provider/manual/environ.go

```diff
@@ -239,26 +239,36 @@ func (e *manualEnviron) DestroyController(controllerUUID string) error {
 	script := `
 set -x
 touch %s
-# If jujud is running, we then wait for a while for it to stop.
+
 stopped=0
-if pkill -%d jujud; then
+function wait_for_jujud {
+    # Give jujud a chance to stop after pkill.
     for i in {1..30}; do
         if pgrep jujud > /dev/null ; then
             sleep 1
         else
-            echo "jujud stopped"
+            echo jujud stopped
             stopped=1
             logger --id jujud stopped on attempt $i
             break
         fi
     done
-fi
-if [ $stopped -ne 1 ]; then
+}
+
+# There might be no jujud at all (for example, after a failed deployment) so
+# don't require pkill to succeed before checking jujud is dead.
+pkill -%d jujud
```

Comment:
would 'don't require it to succeed' actually be better as

pkill ... || true
?